adam/perception/_matcher.py

```diff
@@ -0,0 +1,862 @@
+# -*- coding: utf-8 -*-
```

Comment:
@denizbeser and @LichMaster98 : Since the code in this module is taken largely from NetworkX and is quite complex, you do not need to review it.